include/deal.II/fe/fe_interface_values.h

```diff
@@ -1452,9 +1479,26 @@ namespace FEInterfaceViews
   Scalar<dim, spacedim>::average(const unsigned int interface_dof_index,
                                  const unsigned int q_point) const
   {
-    return this->fe_interface->average(interface_dof_index,
-                                       q_point,
-                                       extractor.component);
+    const auto dof_pair = this->fe_interface->dofmap[interface_dof_index];
+
+    if (this->fe_interface->at_boundary())
+      return 1.0 * (*(this->fe_interface->fe_face_values))[extractor].value(
+                     dof_pair[0], q_point);
```

Comment:
Can you explain why you write `1.0 * (...)` here? The multiplication seems superfluous to me.